storage/s3.go

```diff
@@ -43,6 +46,9 @@ const (
 
 	// Google Cloud Storage endpoint
 	gcsEndpoint = "storage.googleapis.com"
+
+	// the key of the object metadata which is used to handle retry decision on NoSuchUpload error
+	metadataKeyRetryId = "S5cmd--Upload-Retry-Id"
```

Comment:
```suggestion
	metadataKeyRetryId = "s5cmd-upload-retry-id"
```

Reply:
all lowercase.

Reply:
Thanks. The extra hyphen was a typo but capitalization was not a typo.

Metadata fields are stored in lowercase[^lower]. But when the HeadObjectOutput.Metadata is created "By default ... the first letter and any letter following a hyphen will be capitalized, and the rest as lowercase... Set `aws.Config.LowerCaseHeaderMaps` to `true` to write ... as lowercase" [^meta]. So we should either 
1) change the key to match default behaviour as I did
2) or use call "WithLowerCaseHeaderMaps(true)" when the AWS session configuration is created.

In the AWS-SDK v2 this default behaviour is changed[^v2]. So I will convert it to lower case as you've suggested and add the configuration option (which is to be removed. when we migrate to v2).

[^lower]: User-defined metadata is a set of key-value pairs. [Amazon S3 stores user-defined metadata keys in lowercase ](https://docs.aws.amazon.com/AmazonS3/latest/userguide/UsingMetadata.html#:~:text=User%2Ddefined%20metadata%20is%20a%20set%20of%20key%2Dvalue%20pairs.%20Amazon%20S3%20stores%20user%2Ddefined%20metadata%20keys%20in%20lowercase).
[^meta]: https://docs.aws.amazon.com/sdk-for-go/api/service/s3/#HeadObjectOutput
[^v2]: https://pkg.go.dev/github.com/aws/aws-sdk-go-v2/service/s3#HeadObjectOutput.Metadata

Reply:
Good to know, thank you.

Reply:
You're welcome!